synapse/cores/common.py

```diff
@@ -1901,6 +1901,31 @@ def formTufoByTufo(self, tufo):
 
         return self.formTufoByProp(form, valu, **props)
 
+    def formTufosByItems(self, items):
+        '''
+        Forms tufos by prop, given a tuple of (form, valu, props) tuples.
+
+        Args:
+
+            items (tuple): A tuple of tuples of (form, valu, props)
+
+        Returns: tuple of tufos and errors # FIXME not sure how to express this in docs format
+
+        Example:
+
+            items = ( ('foo:thing', 'hehe', {'a': 1}), ('bar:thing', 'haha', {'b': 2}), )
+            results = core.formTufoByItems(items)
+        '''
+        retval = []
+
+        for form, valu, props in items:
+            try:
+                retval.append(self.formTufoByProp(form, valu, **props))
+            except Exception as e:
+                retval.append(e)
```

Comment:
exception objects have to be unwrapped (synerror can help with that) since they can't cross telepath boundaries last i checked.